### ISSUE TITLE
ofxNetwork: UDP Errors

### ISSUE THREAD

**Issue by toliaferrode** [openframeworks/openFrameworks#5645]
I'm testing a UDP application between Arm6 (Raspberry Pi/Raspbian) and Android. I'm using oF 0.9.8 on both Android and the Pi. Everything seems to work fine, it connects, it sends and receives data, but on the Arm6 side of things I always get the error: "[error] ofxNetwork: /home/pi/of_v0.9.8_linuxarm6l_release/addons/ofxNetwork/src/ofxUDPManager,cpp: 248 EINVAL: Invalid argument

It seems to happen if I set the program to send packets in the update function. Maybe I'm not supposed to do that? I need to send data that constantly updates and it seems the update function is the only place to do that in.

I also some times get a similar error message to the one aforementioned but it says host not reachable instead and that's rare. Although the network connection is established and is sending/receiving data just fine.

I'm wondering if this is effecting the performance of my application.

**Comment by bakercp** [comment]
So to clarify, there are no bugs, but you get warnings and errors sometimes? Can you provide some example code to reproduce the errors?

**Comment by arturoc** [comment]
you might be sending to much data, try sending once every 2 or or more frames until you don't see the error

**Comment by toliaferrode** [comment]
@bakercp That's correct. I get the invalid argument error all the time. My code is attached.
[simPrototypeLinux00.tar.gz](https://github.com/openframeworks/openFrameworks/files/1033719/simPrototypeLinux00.tar.gz)

@arturoc Ah, I see. I'm new to network coding. How would you suggest I limit the sending?

**Comment by bakercp** [comment]
No timer needed.  One really easy way would be to just send messages once.

e.g.

```
    if(network.HasSocket() && !sendMessage.empty()){
        network.Send(sendMessage.c_str(), sendMessage.length());
        sendMessage.clear();
    }
```

then when your your mousePressed / mouseReleased update the sendMessage with the number, it will no longer be empty and will be sent once on the next update loop.

**Comment by toliaferrode** [comment]
@bakercp Yes, indeed. Thanks for checking my code. I built this from the UDP sender and receiver example. While that works for this use case, in other cases I need to keep an up to date recording of positional data of of3DPrimitives, and other data. That's why I have it in update().

**Comment by toliaferrode** [comment]
@bakercp @arturoc Oh, I just thought of something. When I test on desktop and Raspberry Pi I don't get any errors I'm pretty sure.
